### PR TITLE
[NPU] Docs op performance optimize

### PR DESCRIPTION
**Pull request by McZyWu**
## Motivation

This is a guidance about how to obtain and analyze profiling data, and thus enhance performance on the operator level.

## Modifications
As follows

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #26232328668](https://github.com/sgl-project/sglang/actions/runs/26232328668)
Latest PR Test (Extra): :x: [Run #26232327436](https://github.com/sgl-project/sglang/actions/runs/26232327436)

### CODE REVIEW

File: docs_new/docs/hardware-platforms/ascend-npus/ascend_npu_operator_performance_optimizing.mdx [hunk 1]
```diff
@@ -0,0 +1,156 @@
+# Operator Performance Optimizing Guidance
+
+# performance_benchmark
```
[the remark this answers is not in the document]
Comment (McZyWu) [reply docs_new/docs/hardware-platforms/ascend-npus/ascend_npu_operator_performance_optimizing.mdx]: Applied. thanks

File: docs_new/docs/hardware-platforms/ascend-npus/ascend_npu_operator_performance_optimizing.mdx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+## Obtaining Performance Data
+
+Before optimizing the performance, you need to obtain accurate performance data, understand the current performance status, and analyze the next optimization direction based on the performance status. MindStudio provides multiple methods for testing the performance of Triton operators, including profiling on the board and single-operator performance simulation pipeline.
+
+### Network
+'''
```
[the remark this answers is not in the document]
Comment (McZyWu) [reply docs_new/docs/hardware-platforms/ascend-npus/ascend_npu_operator_performance_optimizing.mdx]: Applied.

File: docs_new/docs/hardware-platforms/ascend-npus/ascend_npu_operator_performance_optimizing.mdx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        experimental_config=experimental_config) as prof:
+    
+        for step in range(steps):
+            train_one_step(step, steps, train_loader, model, optimizer, criterion)
+            prof.step()
+
+'''
+
+
+Refer to the function event_loop_overlap of this document https://github.com/sgl-project/sglang/blob/ifmn/eagle-dp-attn/python/sglang/srt/managers/scheduler.py to revise the same function in scheduler.py
```
[the remark this answers is not in the document]
Comment (McZyWu) [reply docs_new/docs/hardware-platforms/ascend-npus/ascend_npu_operator_performance_optimizing.mdx]: maybe

File: docs_new/docs/hardware-platforms/ascend-npus/ascend_npu_operator_performance_optimizing.mdx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+
+### Device-end
+The msProf tool is used to collect and analyze key performance indicators of operators running on the Ascend AI Processor. You can use the output performance data to quickly locate the software and hardware performance bottlenecks of operators and improve the efficiency of operator performance analysis.
+
+'''
```
[the remark this answers is not in the document]
Comment (McZyWu) [reply docs_new/docs/hardware-platforms/ascend-npus/ascend_npu_operator_performance_optimizing.mdx]: alright

File: docs_new/docs/hardware-platforms/ascend-npus/ascend_npu_operator_performance_optimizing.mdx [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+    msprof op --kernel-name=xxxxx python3 test_xxxxx.py
+
+'''
+
+'''
```
[the remark this answers is not in the document]
Comment (McZyWu) [reply docs_new/docs/hardware-platforms/ascend-npus/ascend_npu_operator_performance_optimizing.mdx]: ok

File: docs_new/docs/hardware-platforms/ascend-npus/ascend_npu_operator_performance_optimizing.mdx [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+For some operators, the execution time is too long. As a result, the metric data is inaccurate and no longer has reference value. Such data is set to N/A and is not displayed.
+
+Input Shapes set to an empty value indicates that when the format is "; ; ; ;", the current input is a scalar. Here, ";" serves as the separator for each dimension. The output dimension of the operator follows the same principle.
+
+Task Duration (us)
+Time required for running a task, including the time for scheduling a task to the accelerator, execution time on the accelerator, and response end time. The unit is μs.
+Task Wait Time (us)
+Interval between the end time of the previous task and the start time of the current task. The unit is μs.
+Block Dim
+Number of blocks into which a task is divided, which corresponds to the number of cores used for running the task. If task_time is l0, this field is not collected and is displayed as 0.
```
[the remark this answers is not in the document]
Comment (McZyWu) [reply docs_new/docs/hardware-platforms/ascend-npus/ascend_npu_operator_performance_optimizing.mdx]: ok

File: docs_new/docs/hardware-platforms/ascend-npus/ascend_npu_operator_performance_optimizing.mdx [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+Golden rule of tuning: 90% of performance problems are caused by inefficient memory movement.
+3. Characteristics of Ascend instructions
+Good at accessing large contiguous memory blocks
+Dislikes discrete access, stride access, and random access
+Must be 128-bit/256-bit aligned
+Must be vectorized.
+
+## Tips
+1.  Ascend 910 series usually has only 40 or 48 vector cores. If the number of grids exceeds 40 or 48 vector cores, the grids will be delivered in a queue, resulting in a long waiting time. Therefore, the number of cores for high-performance implementation does not exceed the number of vector cores.
+2. Try to use up all the UB as much as possible. Move a large block size at a time to ensure that the bound is in the MTE. No Redundant Copy。
```
[the remark this answers is not in the document]
Comment (McZyWu) [reply docs_new/docs/hardware-platforms/ascend-npus/ascend_npu_operator_performance_optimizing.mdx]: ok

File: docs_new/docs/hardware-platforms/ascend-npus/ascend_npu_operator_performance_optimizing.mdx [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+3. If the offset is a negative number, the current triton-ascend considers it as a discrete memory access scenario. As a result, the performance severely deteriorates, and the data is read from the entire DMA block instead of being read in scalar mode.
+4. The UB of the Ascend hardware requires that the size of the tail axis of the tensor can be exactly divided by 32 bytes. If the length of the tail axis is insufficient, the length of the tail axis is automatically supplemented. For example, the performance deteriorates exponentially due to automatic supplementation for the Tensor whose shape is (2048, 3). In this situation, you can perform the transposition operation to change the alignment axis to a lower dimension. In addition, the transposition operation is affected by the automatic supplement rule. Therefore, special skills are also required to avoid supplementation.
+5. Use Double Buffer, parallelizes computation and data transfer. While computing one block of data, another block of data is being transferred to L1.
```
[the remark this answers is not in the document]
Comment (McZyWu) [reply docs_new/docs/hardware-platforms/ascend-npus/ascend_npu_operator_performance_optimizing.mdx:113]: ok
